src/test/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapperTests.java

```diff
@@ -607,15 +613,18 @@ public void testLuceneFieldMapper_parseCreateField_docValues() throws IOExceptio
         assertArrayEquals(TEST_VECTOR, knnVectorField.vectorValue(), 0.001f);
 
         // Test when doc values are disabled
-        inputBuilder.hasDocValues(false);
-        luceneFieldMapper = createMockLuceneFieldMapper(inputBuilder.build(), TEST_VECTOR);
-
         document = new ParseContext.Document();
         contentPath = new ContentPath();
         parseContext = mock(ParseContext.class);
         when(parseContext.doc()).thenReturn(document);
         when(parseContext.path()).thenReturn(contentPath);
 
+        inputBuilder.hasDocValues(false);
+        luceneFieldMapper = Mockito.spy(new LuceneFieldMapper(inputBuilder.build()));
```

Comment:
why do we need to re-initialize the mapper with the spy()? seems we create same instance, is is to cleanup up mockito state or something else?

Reply:
We need to build a different lucenefieldmapper with docvalues set to false.